_docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme

```diff
@@ -86,113 +87,194 @@ function rhdp2_preprocess_node(&$variables) {
+    if ($variables['view_mode'] === 'compact_dynamic_article_list_item') {
+      $variables['rhd_domain'] = FALSE;
+      $variables['absolute_url'] = FALSE;
+
+      $current_path = \Drupal::service('path.current')->getPath();
+      $url = Url::fromUri("internal:$current_path");
+      $variables['absolute_url'] = $url->setAbsolute()->toString();
+
+      // Requires redhat domain to run comment js.
+      if ($_SERVER['SERVER_NAME']) {
+        $host = parse_url($variables['absolute_url'], PHP_URL_HOST);
+        if ($host === $_SERVER['SERVER_NAME'] && !(strpos($host, 'redhat.com') === FALSE)) {
+          $variables['rhd_domain'] = TRUE;
+        }
       }
     }
   }
 
   // Author content type.
-  if ($type == 'author') {
+  if ($type === 'author') {
     $variables['author_name'] = $node->getTitle();
     $variables['author_articles'] = FALSE;
+    $variables['social_links'] = FALSE;
+
     if (views_get_view_result('author_articles')) {
       $variables['author_articles'] = views_embed_view('author_articles');
     }
-    $variables['social_links'] = FALSE;
+
     $social_facebook = $node->get('field_facebook')->getValue();
     $social_github = $node->get('field_github')->getValue();
     $social_linkedin = $node->get('field_linkedin')->getValue();
     $social_twitter = $node->get('field_twitter')->getValue();
     $social_youtube = $node->get('field_youtube')->getValue();
-    $variables['social_github'] = ($social_github) ? $social_github[0]['uri'] : FALSE;
+
+    // @TOD: Is this meant to be outside of the social_links true logic?
+    $variables['social_github'] = $social_github[0]['uri'] ?? FALSE;
+
     if ($social_facebook || $social_github || $social_linkedin || $social_twitter || $social_youtube) {
       $variables['social_links'] = TRUE;
-      $variables['social_facebook'] = ($social_facebook) ? $social_facebook[0]['uri'] : FALSE;
-      $variables['social_linkedin'] = ($social_linkedin) ? $social_linkedin[0]['uri'] : FALSE;
-      $variables['social_twitter'] = ($social_twitter) ? $social_twitter[0]['uri'] : FALSE;
-      $variables['social_youtube'] = ($social_youtube) ? $social_youtube[0]['uri'] : FALSE;
+      $variables['social_facebook'] = $social_facebook[0]['uri'] ?? FALSE;
+      $variables['social_linkedin'] = $social_linkedin[0]['uri'] ?? FALSE;
+      $variables['social_twitter'] = $social_twitter[0]['uri'] ?? FALSE;
+      $variables['social_youtube'] = $social_youtube[0]['uri'] ?? FALSE;
     }
   }
 
-  // Attach libraries to the Video Resource, default view mode.
-  if ($type === 'video_resource' && $variables['view_mode'] === 'full') {
-    $variables['#attached']['library'][] = 'rhdp2/show-more';
-    $variables['#attached']['library'][] = 'rhdp2/related-content';
+  // Events content type.
+  if ($type == 'events') {
+    $variables['event_title'] = $node->getTitle();
+    $variables['event_description'] = $variables['content']['field_description'][0]['#text'] ?? '';
```

Comment:
This is okay since `field_description` is a required field

Reply:
I see that it is a required field,  however in the case that it is removed from the 'content' output of the view mode, I think it's okay to leave as a ternary.  Please confirm if you'd prefer that I remove the check.